### PR TITLE
Split net::tcp code into files

### PR DESCRIPTION
**Pull request by kpp**
- Incoming -> src/net/tcp/incoming.rs
- TcpListener -> src/net/tcp/listener.rs
- TcpStream, ConnectFuture -> src/net/tcp/stream.rs

### CODE REVIEW

File: src/net/tcp/tcp_stream.rs [hunk 1]
```diff
@@ -254,7 +58,7 @@ impl TcpStream {
         ConnectFuture { inner }
     }
 
-    fn new(connected: mio::net::TcpStream) -> TcpStream {
+    pub(crate) fn new(connected: mio::net::TcpStream) -> TcpStream {
```
Comment (kpp) [inline src/net/tcp/tcp_stream.rs]: Added `pub(crate)` to reuse it in TcpListener:

```
    pub fn poll_accept(&mut self) -> Poll<(TcpStream, SocketAddr), io::Error> {
        let (io, addr) = try_ready!(self.poll_accept_std());

        let io = mio::net::TcpStream::from_stream(io)?;
        let io = PollEvented2::new(io);
        let io = TcpStream { io };

        Ok((io, addr).into())
    }
```

File: src/net/tcp/incoming.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+/// Stream returned by the `TcpListener::incoming` function representing the
+/// stream of sockets received from a listener.
+#[must_use = "streams do nothing unless polled"]
+#[derive(Debug)]
+pub struct Incoming {
+    inner: TcpListener,
+}
+
+impl Incoming {
+    pub(crate) fn with_tcp_listener(listener: TcpListener) -> Incoming {
```
Comment (kpp) [inline src/net/tcp/incoming.rs]: Added to use in TcpListener:
```
    pub fn incoming(self) -> Incoming {
        Incoming::with_tcp_listener(self)
    }
```

**Comment by carllerche** [comment]
What do you think about naming the files `stream.rs` and `listener.rs` to avoid the tcp stutter?

File: src/net/tcp/incoming.rs (hunk 2 above)
Comment (carllerche) [reply src/net/tcp/incoming.rs]: This function can probably just be called `new`.

**Comment by kpp** [comment]
> What do you think about naming the files stream.rs and listener.rs to avoid the tcp stutter?

Done

File: src/net/tcp/incoming.rs (hunk 2 above)
Comment (kpp) [reply src/net/tcp/incoming.rs]: Done

**Comment by carllerche** [comment]
Thanks.

**Pull request merged**